source/adapters/cuda/adapter.cpp

```diff
@@ -0,0 +1,89 @@
+//===--------- adapter.cpp - CUDA Adapter ---------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
```

Comment:
These licenses will need updating too, but a follow up PR is fine.

Reply:
I will add this in a follow up PR